cardano-cli/src/Cardano/CLI/Run/Friendly.hs

```diff
@@ -36,145 +28,139 @@ friendlyTxBodyBS era =
   encodePretty (setConfCompare compare defConfig) . friendlyTxBody era
 
 friendlyTxBody :: CardanoEra era -> TxBody era -> Aeson.Value
-friendlyTxBody era txbody =
-  Object $
-    HashMap.fromList ["era" .= toJSON era]
-    <>
-    case txbody of
-      ByronTxBody body -> friendlyTxBodyByron body
-      ShelleyTxBody ShelleyBasedEraShelley body _scripts _ aux _ ->
-        addAuxData aux $ friendlyTxBodyShelley body
-      ShelleyTxBody ShelleyBasedEraAllegra body _scripts _ aux _ ->
-        addAuxData aux $ friendlyTxBodyAllegra body
-      ShelleyTxBody ShelleyBasedEraMary body _scripts _ aux _ ->
-        addAuxData aux $ friendlyTxBodyMary body
-      ShelleyTxBody ShelleyBasedEraAlonzo _ _ _ _ _ ->
-        panic "friendlyTxBody: Alonzo not implemented yet" -- TODO alonzo
-
-addAuxData :: Show a => Maybe a -> Object -> Object
-addAuxData = HashMap.insert "auxiliary data" . maybe Null (toJSON . textShow)
-
-friendlyTxBodyByron :: Annotated Byron.Tx ByteString -> Object
-friendlyTxBodyByron = assertObject . toJSON
-
-friendlyTxBodyShelley
-  :: Shelley.TxBody (Ledger.ShelleyEra StandardCrypto) -> Object
-friendlyTxBodyShelley body =
-  HashMap.fromList
-    [ "inputs" .= Shelley._inputs body
-    , "outputs" .= fmap friendlyTxOutShelley (Shelley._outputs body)
-    , "certificates" .= fmap textShow (Shelley._certs body)
-    , "withdrawals" .= Shelley.unWdrl (Shelley._wdrls body)
-    , "fee" .= Shelley._txfee body
-    , "time to live" .= Shelley._ttl body
-    , "update" .= fmap textShow (Shelley._txUpdate body)
-    , "metadata hash" .= fmap textShow (Shelley._mdHash body)
-    ]
+friendlyTxBody
+  era
+  (TxBody
+    TxBodyContent
+      { txIns
+      , txOuts
+      , txFee
+      , txValidityRange
+      , txWithdrawals
+      , txCertificates
+      , txUpdateProposal
+      , txMintValue
+      }) =
+  object
+    $   [ "era"     .= era
+        , "fee"     .= friendlyFee txFee
+        , "inputs"  .= friendlyInputs txIns
+        , "outputs" .= map friendlyTxOut txOuts
+        ]
+    ++  [ "certificates" .= friendlyCertificates txCertificates
+        | Just _ <- [certificatesSupportedInEra era]
+        ]
+    ++  [ "mint" .= friendlyMintValue txMintValue
+        | Right _ <- [multiAssetSupportedInEra era]
+        ]
+    ++  [ "update proposal" .= friendlyUpdateProposal txUpdateProposal
```

Comment:
https://github.com/input-output-hk/cardano-node/pull/2613#discussion_r701690475

@cblp There are already many small functions. What I suggest will remove all of the list comprehension noise.